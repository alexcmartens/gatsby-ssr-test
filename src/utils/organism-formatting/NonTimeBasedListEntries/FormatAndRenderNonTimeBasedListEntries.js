import React from "react"
import renderPaddingOrFullBleed from "../renderPaddingOrFullBleed"
import formatPageLinks from "../formatPageLinks"
import { NonTimeBasedListEntries } from "terra-one"

const FormatAndRenderNonTimeBasedListEntries = ({
  data,
  locale,
  paddingClass,
  currentSpaceId = "",
  products,
  siteIdentifier
}) => {
  const entries = []
  let sidebarImage = null

  if (data?.sidebarImage?.image?.file?.url) {
    sidebarImage = {
      focalPointImage: {
        url: data?.sidebarImage?.image?.file?.url,
        height: data?.sidebarImage?.image?.file?.details?.image?.height,
        width: data?.sidebarImage?.image?.file?.details?.image?.width,
        x: data?.sidebarImage?.focalPoint?.focalPoint?.x,
        y: data?.sidebarImage?.focalPoint?.focalPoint?.y,
        alt: data?.sidebarImage?.title,
      },
    }
  }

  if (data?.entries?.length) {
    for (let i = 0; i < data.entries.length; i++) {
      const current = data.entries[i]
      let currentImage = null

      if (current?.image?.image?.file?.url) {
        currentImage = {
          focalPointImage: {
            url: current?.image?.image?.file?.url
              ? current?.image?.image?.file?.url
              : null,
            height: current?.image?.image?.file?.details?.image?.height,
            width: current?.image?.image?.file?.details?.image?.width,
            x: current?.image?.focalPoint?.focalPoint?.x,
            y: current?.image?.focalPoint?.focalPoint?.y,
            alt: current?.image?.title,
          },
        }
      }

      const linkType = current?.link?.url?.contentType
        ? current?.link?.url?.contentType
        : current?.link?.url?.internal?.type
        ? current?.link?.url?.internal?.type
        : null

      const link = formatPageLinks(
        current?.link,
        locale,
        linkType,
        null,
        currentSpaceId,
        products,
        siteIdentifier
      )

      let entryHold = {
        eyebrow: current?.eyebrow,
        header: current?.header,
        subhead: current?.subhead,
        link,
        customAriaLabel: current?.link?.description,
        image: currentImage,
        summary:
          current?.referenceForSummary?.shortSummary?.childMarkdownRemark
            ?.html || current?.shortSummary?.childMarkdownRemark?.html,
        industryCategory: current?.industryOrCategory?.name,
        groupingCategory: current?.groupingCategory?.name,
        role: current?.role,
      }

      entries.push(entryHold)
    }
  }

  return entries?.length
    ? renderPaddingOrFullBleed(
        <NonTimeBasedListEntries
          variant={data?.variant}
          sidebarImage={sidebarImage}
          entries={entries}
          groupEntries={data?.groupEntries}
          paddingClass={paddingClass}
        />
      )
    : null
}

export default FormatAndRenderNonTimeBasedListEntries
