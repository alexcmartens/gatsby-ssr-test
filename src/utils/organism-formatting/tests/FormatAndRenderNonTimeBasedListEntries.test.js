import React from "react"
import renderer from "react-test-renderer"
import FormatAndRenderNonTimeBasedListEntries from "../NonTimeBasedListEntries/FormatAndRenderNonTimeBasedListEntries"

describe("FormatAndRenderNonTimeBasedListEntries", () => {
  it("renders correctly", () => {
    const data = {
      __typename: "ContentfulNonTimeBasedListEntries",
      variant: "long-description",
      groupEntries: null,
      sidebarImage: {
        focalPointImage: {
          height: 4000,
          title: "dog",
          url: "//images.ctfassets.net/citn2sn5tdjr/7wBhLKP8AN67gmNO0MJzsP/151f11bbe5130466fbe942aa63fd42fa/dog.jpg",
          width: 6000,
          x: 0,
          y: 0,
        },
        alt: "Dog",
      },
      entries: [
        {
          eyebrow: "Test Eyebrow",
          header: "Test Generic List Entry 1",
          subhead: "Test Subhead",
          link: {
            url: "/",
            text: "123",
            description: "Test Link Description",
          },
          image: {
            focalPointImage: {
              height: 4000,
              title: "dog",
              url: "//images.ctfassets.net/citn2sn5tdjr/7wBhLKP8AN67gmNO0MJzsP/151f11bbe5130466fbe942aa63fd42fa/dog.jpg",
              width: 6000,
              x: 0,
              y: 0,
            },
            alt: "Dog",
          },
          referenceForSummary: {
            shortSummary: {
              childMarkdownRemark: {
                html: "<p>This is a short summary</p>",
              },
            },
          },
          shortSummary: {
            childMarkdownRemark: {
              html: "<p>Test Short summary</p>",
            },
          },
          industryOrCategory: {
            name: "Test Category",
          },
          groupingCategory: {
            name: "Test Industry",
          },
          role: "Job Title",
        },
        {
          eyebrow: "Test Eyebrow",
          header: "Test Generic List Entry 2",
          subhead: "Test Subhead",
          link: {
            url: "/",
            text: "123",
            description: "Test Link Description",
          },
          image: {
            id: "ebe8d2b6-23d1-5206-8f1a-d9dfac9bb8c0",
            brandfolderAsset: [
              {
                url: "https://assets2.brandfolder.io/bf-boulder-prod/23srvqxgjnh5mkfswxn64893/v/61471532/original/Bkg-Contemporary-Loft-Office-with-Trimble-Globe.jpg",
              },
            ],
            altText: null,
          },
          referenceForSummary: null,
          shortSummary: {
            childMarkdownRemark: {
              html: "<p>Test Short summary</p>",
            },
          },
          industryOrCategory: {
            name: "Test Industry",
          },
          groupingCategory: {
            name: "Test Category",
          },
          role: "Job Title",
        },
      ],
    }

    const tree = renderer
      .create(
        <FormatAndRenderNonTimeBasedListEntries
          data={data}
          paddingClass={"test-padding-class"}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
