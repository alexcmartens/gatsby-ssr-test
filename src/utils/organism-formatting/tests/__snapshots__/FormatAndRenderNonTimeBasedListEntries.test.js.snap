// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormatAndRenderNonTimeBasedListEntries renders correctly 1`] = `
<section
  className="default-layout   mockconstructor
      "
  style={
    Object {
      "backgroundColor": undefined,
      "marginBottom": "",
    }
  }
>
  <section
    className="default-layout__grid 
          
        "
    style={
      Object {
        "backgroundColor": "",
        "marginBottom": "",
      }
    }
  >
    <div
      className="default-layout__children "
    >
      <div
        entries={
          Array [
            Object {
              "customAriaLabel": "Test Link Description",
              "eyebrow": "Test Eyebrow",
              "groupingCategory": "Test Industry",
              "header": "Test Generic List Entry 1",
              "image": null,
              "industryCategory": "Test Category",
              "link": Object {
                "Link": [MockFunction],
                "iconImage": undefined,
                "override": null,
                "setOverride": null,
                "text": "123",
                "url": "/en/undefined",
                "video": null,
              },
              "role": "Job Title",
              "subhead": "Test Subhead",
              "summary": "<p>This is a short summary</p>",
            },
            Object {
              "customAriaLabel": "Test Link Description",
              "eyebrow": "Test Eyebrow",
              "groupingCategory": "Test Category",
              "header": "Test Generic List Entry 2",
              "image": null,
              "industryCategory": "Test Industry",
              "link": Object {
                "Link": [MockFunction],
                "iconImage": undefined,
                "override": null,
                "setOverride": null,
                "text": "123",
                "url": "/en/undefined",
                "video": null,
              },
              "role": "Job Title",
              "subhead": "Test Subhead",
              "summary": "<p>Test Short summary</p>",
            },
          ]
        }
        groupEntries={null}
        paddingClass="test-padding-class"
        sidebarImage={null}
        variant="long-description"
      />
    </div>
  </section>
</section>
`;
